File: app/src/main/java/cn/intret/app/picgo/model/UserInitialPreferences.java
package cn.intret.app.picgo.model;

import java.util.List;

/**
 * 用户初始偏好设置，包含 UI 决定如何进行首次显示的偏好设置。
 */
public class UserInitialPreferences {
    List<RecentRecord> mRecentRecords;
    ViewMode mViewMode;
    SortWay mSortWay;
    SortOrder mSortOrder;

    public List<RecentRecord> getRecentRecords() {
        return mRecentRecords;
    }

    public UserInitialPreferences setRecentRecords(List<RecentRecord> recentRecords) {
        mRecentRecords = recentRecords;
        return this;
    }

    public ViewMode getViewMode() {
        return mViewMode;
    }

    public UserInitialPreferences setViewMode(ViewMode viewMode) {
        mViewMode = viewMode;
        return this;
    }

    public SortWay getSortWay() {
        return mSortWay;
    }

    public UserInitialPreferences setSortWay(SortWay sortWay) {
        mSortWay = sortWay;
        return this;
    }

    public SortOrder getSortOrder() {
        return mSortOrder;
    }

    public UserInitialPreferences setSortOrder(SortOrder sortOrder) {
        mSortOrder = sortOrder;
        return this;
    }
}

File: app/src/main/java/cn/intret/app/picgo/model/user/data/UserInitialPreferences.kt
package cn.intret.app.picgo.model.user.data

/**
 * 用户初始偏好设置，包含 app 使用最后状态的 UI 布局偏好设置。
 */
class UserInitialPreferences {
    lateinit var recentRecords: List<RecentRecord>
    var viewMode: ViewMode = ViewMode.UNKNOWN
    var sortWay: SortWay = SortWay.UNKNOWN
    var sortOrder: SortOrder = SortOrder.UNKNOWN

    override fun toString(): String {
        return "UserInitialPreferences{" +
                "recentRecords=" + recentRecords +
                ", viewMode=" + viewMode +
                ", sortWay=" + sortWay +
                ", sortOrder=" + sortOrder +
                '}'.toString()
    }
}
